lib/auth/tls_test.go

```diff
@@ -1720,6 +1720,32 @@ func TestGetCertAuthority(t *testing.T) {
 	}, true)
 	require.True(t, trace.IsAccessDenied(err))
 
+	// generate server keys for proxy
+	proxyClt, err := tt.server.NewClient(TestIdentity{I: BuiltinRole{Username: "00000000-0000-0000-0000-000000000001", Role: types.RoleProxy}})
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, proxyClt.Close()) })
+
+	// proxy can't fetch the host CA with secrets
+	_, err = proxyClt.GetCertAuthority(ctx, types.CertAuthID{
+		DomainName: tt.server.ClusterName(),
+		Type:       types.HostCA,
+	}, true)
+	require.Error(t, err)
+
+	// proxy can fetch SAML IdP CA with secrets
```

Comment:
Is this authorized by the change in permissions.go?

Could you briefly explain why the proxy needs secrets in this case, to the uninitiated?

Reply:
This is a brand new feature, but basically the SAML IdP is going to be served out of the proxy. So in this case, the proxy needs to be able to read the SAML IdP's private key so that it can use it for signing of communications between SAML endpoints.

Reply:
Got it, thanks for the explanation.